nixos/lib/systemd-unit-options.nix

```diff
@@ -284,71 +284,107 @@ in rec {
       '';
     };
 
-  }; };
+    script = mkOption {
+      type = types.lines;
+      default = "";
+      description = "Shell commands executed as the service's main process.";
+    };
 
-  stage2ServiceOptions = { name, config, ... }: {
-    imports = [
-      stage2CommonUnitOptions
-      serviceOptions
-    ];
+    scriptArgs = mkOption {
+      type = types.str;
+      default = "";
+      description = "Arguments passed to the main process script.";
+    };
 
-    options = {
-      script = mkOption {
-        type = types.lines;
-        default = "";
-        description = "Shell commands executed as the service's main process.";
-      };
+    preStart = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Shell commands executed before the service's main process
+        is started.
+      '';
+    };
 
-      scriptArgs = mkOption {
-        type = types.str;
-        default = "";
-        description = "Arguments passed to the main process script.";
-      };
+    postStart = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Shell commands executed after the service's main process
+        is started.
+      '';
+    };
 
-      preStart = mkOption {
-        type = types.lines;
-        default = "";
-        description = ''
-          Shell commands executed before the service's main process
-          is started.
-        '';
-      };
+    reload = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Shell commands executed when the service's main process
+        is reloaded.
+      '';
+    };
 
-      postStart = mkOption {
-        type = types.lines;
-        default = "";
-        description = ''
-          Shell commands executed after the service's main process
-          is started.
-        '';
-      };
+    preStop = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Shell commands executed to stop the service.
+      '';
+    };
 
-      reload = mkOption {
-        type = types.lines;
-        default = "";
-        description = ''
-          Shell commands executed when the service's main process
-          is reloaded.
-        '';
-      };
+    postStop = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Shell commands executed after the service's main process
+        has exited.
+      '';
+    };
 
-      preStop = mkOption {
-        type = types.lines;
-        default = "";
-        description = ''
-          Shell commands executed to stop the service.
-        '';
-      };
+    jobScripts = mkOption {
+      type = with types; coercedTo path singleton (listOf path);
+      internal = true;
+      description = "A list of all job script derivations of this unit";
+      default = [];
+    };
```

Comment:
"A list of all job script derivations" doesn't seem useful, whereas "a list of store paths to include for this unit" does.
How about `extraInitrdPaths`? (or however you camelcase `initrd`)
Might want to reserve `extraInitrdStorePaths` for adding whole closures, as _store path_ generally has the meaning of the whole subdirectory under the storeDir.

Reply:
It's actually also set in stage 2 units but never used. I don't know, maybe there's some use case for that in the future? Apparmor or whatnot?

Reply:
I guess those could add to `system.extraDependencies`, but it seems a little pointless.

It's ok to leave this as an implementation detail. It's `internal`.